src/main/kotlin/com/slack/moshi/interop/gson/InteropBuilder.kt

```diff
@@ -245,3 +255,39 @@ private fun JsonElement.toJsonValue(): Any? {
     else -> error("Not possible")
   }
 }
+
+@Suppress("ComplexMethod") // It's not too complex, Detekt
+private fun Any?.toJsonElement(): JsonElement {
+  return when (this) {
+    null -> JsonNull.INSTANCE
+    is Boolean -> JsonPrimitive(this)
+    is Char -> JsonPrimitive(this)
+    is Number -> JsonPrimitive(this)
+    is String -> JsonPrimitive(this)
+    is Array<*> -> {
+      JsonArray(size).apply {
+        for (element in this) {
+          add(element.toJsonElement())
+        }
+      }
+    }
+    is Collection<*> -> {
+      JsonArray(size).apply {
+        for (element in this@toJsonElement) {
```

Comment:
Tricky

Reply:
I scratched my head at this for longer than I'm willing to admit

Reply:
yay for tests